tests/Unit/Evolution/Systems/GrMhd/ValenciaDivClean/FiniteDifference/Test_Wcns5z.cpp

```diff
@@ -0,0 +1,33 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include "Evolution/Systems/GrMhd/ValenciaDivClean/FiniteDifference/Factory.hpp"
+#include "Evolution/Systems/GrMhd/ValenciaDivClean/FiniteDifference/Tag.hpp"
+#include "Evolution/Systems/GrMhd/ValenciaDivClean/FiniteDifference/Wcns5z.hpp"
+#include "Framework/TestCreation.hpp"
+#include "Helpers/Evolution/Systems/GrMhd/ValenciaDivClean/FiniteDifference/PrimReconstructor.hpp"
+
+SPECTRE_TEST_CASE("Unit.Evolution.Systems.GrMhd.ValenciaDivClean.Fd.Wcns5zPrim",
+                  "[Unit][Evolution]") {
+  namespace helpers = TestHelpers::grmhd::ValenciaDivClean::fd;
+  const grmhd::ValenciaDivClean::fd::Wcns5zPrim wcns5z_recons{2, 2.0e-16};
+  helpers::test_prim_reconstructor(5, wcns5z_recons);
+
+  const auto wcns5z_from_options_base = TestHelpers::test_factory_creation<
+      grmhd::ValenciaDivClean::fd::Reconstructor,
+      grmhd::ValenciaDivClean::fd::OptionTags::Reconstructor>(
+      "Wcns5zPrim:\n"
+      "  NonlinearWeightExponent: 2\n"
+      "  Epsilon: 2.0e-16\n");
```

Comment:
~you can add a `->get_clone()` here and a `serialize_and_deserialize(TestHelpers...->get_clone());` I think that'll take care of pretty much all the codecov things. We should probably do this for all the reconstructors. If you want, feel free to add a commit to this PR that does that, or one of us can just do it later as a separate PR :)~

I think the "right" way of doing this is to change the `test_prim_reconstructor` function:
```cpp
                              const Reconstructor& in_derived_reconstructor) {
PUPable_reg(Reconstructor);
const auto base_recons = serialize_and_deserialize(in_derived_reconstructor.get_clone());
const auto& derived_reconstructor = dynamic_cast<const Reconstructor&>(*base_recons);
```